### PR TITLE
Fix a bug in ImagickDriver::pixelate that caused a DivideByZero excep…

### PR DESCRIPTION
**Pull request by psion-ar**
Hi there :)

Line 477 in ImagickDriver caused a DivideByZero exception.

```console

   FAIL  Tests\Manipulations\PixelateTest
  ⨯ it can pixelate an image with dataset "imagick" / (0)                                                                                                                                   0.02s  
  ✓ it can pixelate an image with dataset "imagick" / (50)                                                                                                                                  0.16s  
  ✓ it can pixelate an image with dataset "imagick" / (100)                                                                                                                                 0.12s  
  ✓ it can pixelate an image with dataset "gd" / (0)                                                                                                                                        0.11s  
  ✓ it can pixelate an image with dataset "gd" / (50)                                                                                                                                       0.12s  
  ✓ it can pixelate an image with dataset "gd" / (100)                                                                                                                                      0.12s  
  ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────  
  FAILED  Tests\Manipulations\PixelateTest > it can pixelate an image with dataset "imagick" / (0)                                                                            DivisionByZeroError   
  Division by zero

  at src/Drivers/Imagick/ImagickDriver.php:477
    473▕         $width = $this->getWidth();
    474▕         $height = $this->getHeight();
    475▕ 
    476▕         foreach ($this->image as $image) {
  ➜ 477▕             $image->scaleImage(max(1, (int) ($width / $pixelate)), max(1, (int) ($height / $pixelate)));
    478▕             $image->scaleImage($width, $height);
    479▕         }
    480▕ 
    481▕         return $this;

  1   src/Drivers/Imagick/ImagickDriver.php:477
  2   src/Image.php:266


  Tests:    1 failed, 5 passed (5 assertions)
  Duration: 0.74s
```

I just wrapped the pixelate logic in a conditional, to run only when `$pixelate !== 0`.  
The method signatures for all 3 "effect" methods that require argument values remain the same.  
The tests are updated (BlurTest and SharpenTest too) to cover edge cases and to  
stay consistent across the three method's i mentioned.

btw... <3

Regard's Alex

p.s. dont know why the rotation snapshots got updated. didnt mess with those

### CODE REVIEW

File: src/Drivers/Imagick/ImagickDriver.php [hunk 1]
```diff
@@ -470,12 +470,14 @@ public function flip(FlipDirection $flip): static
 
     public function pixelate(int $pixelate = 50): static
     {
-        $width = $this->getWidth();
-        $height = $this->getHeight();
+        if ($pixelate !== 0) {
```
Comment (freekmurze) [inline src/Drivers/Imagick/ImagickDriver.php:473]: Let's check if `$pixelate` is `0` and then to an early return. This way, we'll lose a level of indentation.

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: src/Drivers/Imagick/ImagickDriver.php (hunk 1 above)
Comment (psion-ar) [reply src/Drivers/Imagick/ImagickDriver.php]: ok new commit... hope i did not mess up.
(first contribution ever:x)

**Comment by freekmurze** [comment]
This PR contains a lot of stylistic changes (how `null` is checked, strict type checking). 

Could you remove those change and only change the bare necessities?

**Comment by psion-ar** [comment]
Of course. Would it perhaps be best to close the pr and open a new one? sorry if I'm causing trouble;x
